face/src/main/java/io/blockv/face/client/FaceView.kt

```diff
@@ -5,27 +5,67 @@ import android.view.View
 import android.view.ViewGroup
 import io.blockv.common.model.Face
 import io.blockv.common.model.Vatom
+import io.blockv.common.util.Cancellable
+import io.blockv.common.util.CompositeCancellable
 
 
 abstract class FaceView(
   var vatom: Vatom,
   var face: Face,
   val bridge: FaceBridge
 ) {
+  private var cancellable: CompositeCancellable
+
+  @get:Synchronized
+  @set:Synchronized
+  internal var isLoaded: Boolean = false
+
+  init {
+    cancellable = CompositeCancellable()
+  }
 
   abstract fun onCreateView(inflater: LayoutInflater, container: ViewGroup): View
 
   abstract fun onLoad(handler: LoadHandler)
 
-  abstract fun onVatomUpdate(vatom: Vatom)
+  fun update(vatom: Vatom, handler: LoadHandler) {
+    val temp = this.vatom
+    onVatomChanged(vatom)
+    this.vatom = vatom
+    if (onVatomChanged(temp, vatom)) {
+      onLoad(handler)
+    } else
+      handler.onComplete()
+  }
 
-  abstract fun onUnload()
+  open fun onVatomChanged(updatedVatom: Vatom) {}
```

Comment:
should the first argument rather be `changedVatom`?

Reply:
I would prefer newVatom

Reply:
Sure, `vatom` is good.